srml/balances/README.adoc

```diff
@@ -53,32 +120,44 @@ let ed = Balances::existential_deposit();
 let tf = Balances::transfer_fee();
 ```
 
-3. Include a balance transfer in a block (example from the `executor` module tests):
+==== Real Use Example
+
+Use a balance transfer to buy a link:https://github.com/shawntabrizi/substrate-collectables-workshop/blob/master/3/assets/3.5-finished-code.rs#L105[SubstrateKitty]:
 
 ```rust
-let mut t = new_test_ext(COMPACT_CODE, false);
-let block1 = construct_block(
-    &mut t,
-    1,
-    GENESIS_HASH.into(),
-    vec![
-        CheckedExtrinsic {
-            signed: None,
-            function: Call::Timestamp(timestamp::Call::set(42)),
-        },
-        CheckedExtrinsic {
-            signed: Some((alice(), 0)),
-            function: Call::Balances(balances::Call::transfer(bob().into(), 69)),
-        },
-    ]
-);
+fn buy_kitty(origin, kitty_id: T::Hash, max_price: T::Balance) -> Result {
+    let sender = ensure_signed(origin)?;
+
+    ensure!(<Kitties<T>>::exists(kitty_id), "This kitty does not exist");
+
+    let owner = Self::owner_of(kitty_id).ok_or("No owner for this kitty")?;
+    ensure!(owner != sender, "You can't buy your own kitty");
+
+    let mut kitty = Self::kitty(kitty_id);
+
+    let kitty_price = kitty.price;
+    ensure!(!kitty_price.is_zero(), "The kitty you want to buy is not for sale");
+    ensure!(kitty_price <= max_price, "The kitty you want to buy costs more than your max price");
+
+    // Make a balance transfer
+    <balances::Module<T>>::make_transfer(&sender, &owner, kitty_price)?;
+
+    Self::_transfer_from(owner.clone(), sender.clone(), kitty_id)?;
```

Comment:
if previous call succeeded, this call should never return Err?

Reply:
@xlc good catch. The first two checks are checked before calling the function.
```
let owner = Self::owner_of(kitty_id).ok_or("No owner for this kitty")?;
ensure!(owner == from, "'from' account does not own this kitty");
```
But the overflow checks are not:
```
let new_owned_kitty_count_to = owned_kitty_count_to.checked_add(1)
    .ok_or("Transfer causes overflow of 'to' kitty balance")?;

let new_owned_kitty_count_from = owned_kitty_count_from.checked_sub(1)
    .ok_or("Transfer causes underflow of 'from' kitty balance")?;
```
@shawntabrizi if someone owns `u64::max_value()` kitties and bought another one, he would transfer his money away and not receive his kitty. I can push I fix to the kitties workshop.

Reply:
Example changed, resolving